elk/utils_evaluation/ccs.py

```diff
@@ -11,47 +13,27 @@ def __init__(
         num_epochs=1000,
         num_tries=10,
         learning_rate=1e-2,
+        weight_decay=0.01,
+        optimizer: Literal["adam", "lbfgs"] = "adam",
         device="cuda",
     ):
         self.include_bias = include_bias
         self.verbose = verbose
         self.num_epochs = num_epochs
         self.num_tries = num_tries
         self.learning_rate = learning_rate
+        self.weight_decay = weight_decay
+        self.optimizer = optimizer
         self.device = device
 
-    def init_parameters(self):
+    def init_probe(self):
         """
-        Initializes the parameters of the model.
+        Initializes the probe of the model.
 
         Returns:
-            numpy.ndarray: The initialized parameters of the model.
+            Probe: The initialized probe of the model.
         """
-        init_theta = np.random.randn(self.d).reshape(1, -1)
-        init_theta = init_theta / np.linalg.norm(init_theta)
-        return init_theta
```

Comment:
I think we should keep the initialization of theta for now (at least for Adam). Just for reproducing the actual results of the paper. It could be of course that it doesn't make any difference, and we remove it later.

Reply:
Agreed, I've added a probe which is initialized just like the original CCS. Thanks for catching that.